modules/redshift-loader/src/test/scala/com/snowplowanalytics/snowplow/rdbloader/loading/RedshiftLoaderSpec.scala

```diff
@@ -0,0 +1,279 @@
+import com.snowplowanalytics.snowplow.rdbloader.common.S3
+import com.snowplowanalytics.snowplow.rdbloader.discovery.{DataDiscovery, ShreddedType}
+import com.snowplowanalytics.snowplow.rdbloader.common.config.Semver
+import com.snowplowanalytics.snowplow.rdbloader.common.config.ShredderConfig.Compression
+import com.snowplowanalytics.snowplow.rdbloader.SpecHelpers.AsSql
+import com.snowplowanalytics.snowplow.rdbloader.config.components.PasswordConfig
+import com.snowplowanalytics.snowplow.rdbloader.test.TestState.LogEntry
+import com.snowplowanalytics.snowplow.rdbloader.test._
+import org.specs2.mutable.Specification
+
+class RedshiftLoaderSpec extends Specification {
+  import RedshiftLoaderSpec._
+
+  implicit val dao: RsDao[Pure] = PureDAO.interpreter(PureDAO.init)
+
+  "loadFolder" should {
+    "perform atomic and shredded insertions, ignore VACUUM and ANALYZE" >> {
+      val builder: RedshiftLoader[Pure] = new RedshiftLoader[Pure](tgt, "eu-central-1")
+      val input = RedshiftStatements(
+        "atomic",
+        Statement
+          .EventsCopy("atomic", false, "s3://bucket/path/run=1/".dir, "eu-central-1", 1, "role", Compression.None),
+        List(
+          Statement.ShreddedCopy(
+            "atomic",
+            ShreddedType.Tabular(
+              ShreddedType.Info("s3://bucket/path/run=1/".dir, "com.acme", "context", 1, Semver(1, 0, 0))
+            ),
+            "eu-central-1",
+            1,
+            "role",
+            Compression.None
+          )
+        )
+      )
+
+      val (state, result) = builder.loadFolder(input).run
+
+      val expected = List(
+        LogEntry.Message("setStage copying into events table"),
+        LogEntry.Message("COPY atomic.events"),
+        LogEntry.Message(
+          Statement
+            .EventsCopy("atomic", false, "s3://bucket/path/run=1/".dir, "eu-central-1", 1, "role", Compression.None)
+            .toFragment
+            .toString()
+        ),
+        LogEntry.Message(
+          "COPY atomic.com_acme_context_1 FROM s3://bucket/path/run=1/output=good/vendor=com.acme/name=context/format=tsv/model=1"
+        ),
+        LogEntry.Message("setStage copying into com_acme_context_1 table"),
+        LogEntry.Message(
+          Statement
+            .ShreddedCopy(
+              "atomic",
+              ShreddedType.Tabular(
+                ShreddedType.Info("s3://bucket/path/run=1/".dir, "com.acme", "context", 1, Semver(1, 0, 0))
+              ),
+              "eu-central-1",
+              1,
+              "role",
+              Compression.None
+            )
+            .toFragment
+            .toString()
+        )
+      )
+
+      val transactionsExpectation = state.getLog must beEqualTo(expected)
+      val resultExpectation       = result must beRight
+      transactionsExpectation.and(resultExpectation)
+    }
+
+    "perform atomic transit load with shredded types, ignore VACUUM and ANALYZE" >> {
```

Comment:
Only difference between this test and the one above is being transient. Maybe we might unify them somehow to share common parts.